swap_database.yml

```diff
@@ -0,0 +1,96 @@
+---
+# This will rename a database to swap with an existing one.
+# There will be some downtime during stop services, drop, rename and start services.
+
+- name: get database details
+  hosts: database
+  vars_prompt:
+    - name: "origin_db_name"
+      prompt: "What database are we moving?"
+      private: no
+      default: repository_new
+    - name: "target_db_name"
+      prompt: "What database are we swapping out to (<name>_old)?"
+      private: no
+      default: repository
+  tasks:
+    - set_fact:
+        origin_db_name: "{{ origin_db_name }}"
+        target_db_name: "{{ target_db_name }}"
+        dest_db_name: "{{ target_db_name }}_old"
+
+- name: stop archive
+  hosts: archive
+  tasks:
+    - shell: supervisorctl stop archive
+      become: yes
+
+- name: stop publishing
+  hosts: publishing
+  tasks:
+    - shell: supervisorctl stop publishing
+      become: yes
+
+- name: stop post publication worker
+  hosts: publishing
+  tasks:
+    - shell: supervisorctl stop post_publication
+      become: yes
+
+- name: stop zclients
+  hosts:
+    - zclient
+    - pdf_gen
+  tasks:
+    - shell: "for i in {0..{{ hostvars[inventory_hostname].zclient_count|default(1) -1 }}}; do supervisorctl stop zclient_instance$i; done"
+      become: yes
+      args:
+        executable: /bin/bash
+
+# Varnish is still running and the frontend should be serving stale results.
+
+- name: rename database
+  hosts: database
+  tasks:
+    - name: rename via pgsql
+      shell: |
+        psql -v ON_ERROR_STOP=1 --username postgres <<-EOSQL
+        ALTER DATABASE {{ target_db_name }} RENAME TO {{ dest_db_name }};
+        ALTER DATABASE {{ origin_db_name }} RENAME TO {{ target_db_name }};
+        EOSQL
```

Comment:
Getting this error:

```
TASK [rename via pgsql] ********************************************************
task path: /home/karen/src/cnx-deploy/swap_database.yml:55
fatal: [local.cnx.org]: FAILED! => {"changed": true, "cmd": "psql -v ON_ERROR_STOP=1 --username postgres <<-EOSQL\n ALTER DATABASE cnxarchive RENAME TO cnxarchive_old;\n ALTER DATABASE cnxarchive_new RENAME TO cnxarchive;\n EOSQL", "delta": "0:00:00.079172", "end": "2016-09-14 09:13:59.003535", "failed": true, "rc": 3, "start": "2016-09-14 09:13:58.924363", "stderr": "ERROR:  syntax error at or near \"EOSQL\"\nLINE 1: EOSQL\n        ^", "stdout": "ALTER DATABASE\nALTER DATABASE", "stdout_lines": ["ALTER DATABASE", "ALTER DATABASE"], "warnings": []}
```

Unforunately ansible doesn't seem to handle here doc very well. (https://github.com/ansible/ansible/issues/8512 towards the end)

One of the workarounds suggested is: `<<-" EOSQL"` (a leading space in the here doc limit string)